cabal-install/src/Distribution/Client/CmdRepl.hs

```diff
@@ -451,6 +451,23 @@ replAction flags@NixStyleFlags{extraFlags = r@ReplFlags{..}, ..} targetStrings g
         -- unit files for components
         unit_files <- listDirectory dir
 
+        -- Order the unit files so that the last target becomes the active unit
+        let active_unit_fp :: Maybe FilePath
+            active_unit_fp = do
+              -- Get the first target selectors from the cli
+              activeTarget <- safeHead targetSelectors
+              -- Lookup the targets :: Map UnitId [(ComponentTarget, NonEmpty TargetSelector)]
+              Map.toList targets
+                -- Keep the UnitId matching the desired target selector
+                & filter (\(_, xs) -> any (\(_, selectors) -> activeTarget `elem` selectors) xs)
+                -- Convert to filename
```

Comment:
Is this a safe way to do this? I have no idea. Maybe there is a conversion function exported somewhere?